drivers/entropy/entropy_rpi_pico.c

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2024 Xudong Zheng
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <string.h>
+#include <pico/rand.h>
+#include <zephyr/drivers/entropy.h>
+#include <zephyr/spinlock.h>
+
+#define DT_DRV_COMPAT raspberrypi_pico_rng
+
+static struct k_spinlock entropy_lock;
+
+static int entropy_rpi_pico_get_entropy(const struct device *dev, uint8_t *buf, uint16_t len)
+{
+	k_spinlock_key_t key = k_spin_lock(&entropy_lock);
+
+	__ASSERT_NO_MSG(buf != NULL);
+	uint8_t *buf_bytes = buf;
+
+	while (len > 0) {
+		uint64_t value = get_rand_64();
```

Comment:
```suggestion
		K_SPINLOCK(&entropy_lock) {
			uint64_t value = get_rand_64();
		}
```
Only the call to `get_rand_64()` needs wrapping in the spinlock. Reducing the amount of code in the lock is preferable, and use the helper macro to limit the scope of the key.

Reply:
I've wrapped `get_rand_64()` with the `K_SPINLOCK()` macro.